diff -uraN lftp-4.9.2/configure.ac omv-lftp-4.9.2/configure.ac
--- lftp-4.9.2/configure.ac	2018-12-26 18:14:57.000000000 +0100
+++ omv-lftp-4.9.2/configure.ac	2020-08-14 17:07:09.799388741 +0200
@@ -185,7 +185,7 @@
 	 with_socksdante=yes;;
 esac
 
-if test x$socks_loc != x; then
+if test x$socks_loc != x -a x$socks_loc != /usr; then
    LDFLAGS="$LDFLAGS -L$socks_loc/lib"
    CPPFLAGS="$CPPFLAGS -I$socks_loc/include"
 fi
@@ -306,9 +306,11 @@
 
 AX_CHECK_ZLIB([
    AC_SUBST([ZLIB],[-lz])
+   if [ "${ZLIB_HOME}" != "/usr" ]; then
    r=""; test "$enable_rpath" = yes -a "$ZLIB_HOME" != /usr && r=" -R${ZLIB_HOME}/lib"
    AC_SUBST([ZLIB_LDFLAGS],["-L${ZLIB_HOME}/lib$r"])
    AC_SUBST([ZLIB_CPPFLAGS],["-I${ZLIB_HOME}/include"])
+   fi
 ],[
    AC_MSG_ERROR([cannot find -lz library, install zlib-devel package])
 ])
diff -uraN lftp-4.9.2/m4/ax_check_zlib.m4 omv-lftp-4.9.2/m4/ax_check_zlib.m4
--- lftp-4.9.2/m4/ax_check_zlib.m4	2017-08-04 16:01:33.000000000 +0200
+++ omv-lftp-4.9.2/m4/ax_check_zlib.m4	2020-08-14 17:07:55.160100147 +0200
@@ -104,7 +104,7 @@
 
   ZLIB_OLD_LDFLAGS="$LDFLAGS"
   ZLIB_OLD_CPPFLAGS="$CPPFLAGS"
-  if test -n "${ZLIB_HOME}"; then
+  if test -n "${ZLIB_HOME}" -a "${ZLIB_HOME}" != "/usr"; then
         LDFLAGS="$LDFLAGS -L${ZLIB_HOME}/lib"
         CPPFLAGS="$CPPFLAGS -I${ZLIB_HOME}/include"
   fi
diff -uraN lftp-4.9.2/m4/ax_lib_expat.m4 omv-lftp-4.9.2/m4/ax_lib_expat.m4
--- lftp-4.9.2/m4/ax_lib_expat.m4	2017-08-04 16:01:58.000000000 +0200
+++ omv-lftp-4.9.2/m4/ax_lib_expat.m4	2020-08-14 17:08:38.394966649 +0200
@@ -108,8 +108,10 @@
     run_expat_test="no"
 
     if test -n "$expat_prefix"; then
+        if "$expat_prefix" != "/usr"; then
         expat_include_dir="$expat_prefix/include"
         expat_ld_flags="-L$expat_prefix/lib"
+        fi
         expat_lib_flags="-lexpat"
         run_expat_test="yes"
     elif test "$expat_requested" = "yes"; then
diff -uraN lftp-4.9.2/m4/lftp_lib_readline.m4 omv-lftp-4.9.2/m4/lftp_lib_readline.m4
--- lftp-4.9.2/m4/lftp_lib_readline.m4	2020-01-11 17:02:43.000000000 +0100
+++ omv-lftp-4.9.2/m4/lftp_lib_readline.m4	2020-08-14 17:05:46.201452568 +0200
@@ -107,7 +107,7 @@
 	if test -f "$readline_include_dir/readline/readline.h"; then
 	    readline_include_dir="$readline_include_dir/readline"
 	fi
-        readline_ld_flags="-L$readline_prefix/lib"
+        if [ "$readline_prefix" != "/usr" ]; then readline_ld_flags="-L$readline_prefix/lib"; fi
         if test -z "$readline_lib_flags"; then
             readline_lib_flags="-lreadline"
         fi
diff -uraN lftp-4.9.2/m4/lftp.m4 omv-lftp-4.9.2/m4/lftp.m4
--- lftp-4.9.2/m4/lftp.m4	2017-08-10 08:12:03.000000000 +0200
+++ omv-lftp-4.9.2/m4/lftp.m4	2020-08-14 17:04:51.966388932 +0200
@@ -318,7 +318,7 @@
      AC_ARG_WITH(libidn2, AS_HELP_STRING([--with-libidn2@<:@=DIR@:>@],[Support IDN2 (needs GNU Libidn2)]),
        libidn2=$withval, libidn2=yes)
      if test "$libidn2" != "no"; then
-       if test "$libidn2" != "yes"; then
+       if test "$libidn2" != "yes" -a "$libidn2" != "/usr"; then
          LDFLAGS="${LDFLAGS} -L$libidn2/lib"
          CPPFLAGS="${CPPFLAGS} -I$libidn2/include"
        fi
diff -uraN lftp-4.9.2/m4/ssl.m4 omv-lftp-4.9.2/m4/ssl.m4
--- lftp-4.9.2/m4/ssl.m4	2017-08-04 16:00:40.000000000 +0200
+++ omv-lftp-4.9.2/m4/ssl.m4	2020-08-14 17:09:40.224616924 +0200
@@ -9,7 +9,7 @@
 		old_LDFLAGS="$LDFLAGS"
 		old_CPPFLAGS="$CPPFLAGS"
 		LIBS="$LIBS -lssl -lcrypto"
-		if test $loc != default; then
+		if test $loc != default -a "$loc" != "/usr"; then
 			LDFLAGS="$LDFLAGS -L$loc/lib"
 			CPPFLAGS="$CPPFLAGS -I$loc/include"
 		fi
@@ -20,7 +20,7 @@
 		CPPFLAGS="$old_CPPFLAGS"
 		if test $found_loc != "none"; then
 			lftp_cv_openssl="OPENSSL_LIBS=\"-lssl -lcrypto\""
-			if test $found_loc != default; then
+			if test $found_loc != default -a $found_loc != /usr; then
 				r=""; test "$enable_rpath" = yes -a "$found_loc" != /usr && r=" -R$found_loc/lib"
 				lftp_cv_openssl="$lftp_cv_openssl OPENSSL_LDFLAGS=\"-L$found_loc/lib$r\""
 				lftp_cv_openssl="$lftp_cv_openssl OPENSSL_CPPFLAGS=-I$found_loc/include"
