diff -up lftp-4.8.4/lib/xalloc.h.omv~ lftp-4.8.4/lib/xalloc.h
diff -up lftp-4.8.4/lib/xalloc-oversized.h.omv~ lftp-4.8.4/lib/xalloc-oversized.h
--- lftp-4.8.4/lib/xalloc-oversized.h.omv~	2018-08-22 23:33:50.743554537 +0200
+++ lftp-4.8.4/lib/xalloc-oversized.h	2018-08-22 23:34:09.882640257 +0200
@@ -41,7 +41,7 @@ typedef size_t __xalloc_count_type;
    positive and N must be nonnegative.  This is a macro, not a
    function, so that it works correctly even when SIZE_MAX < N.  */
 
-#if 7 <= __GNUC__
+#if !defined(__clang__) && 7 <= __GNUC__
 # define xalloc_oversized(n, s) \
    __builtin_mul_overflow_p (n, s, (__xalloc_count_type) 1)
 #elif 5 <= __GNUC__ && !defined __ICC && !__STRICT_ANSI__
